src/languageServerInstaller.ts

```diff
@@ -10,9 +10,14 @@ import { Release, getRelease } from '@hashicorp/js-releases';
 
 const extensionVersion = vscode.extensions.getExtension('hashicorp.terraform').packageJSON.version;
 
+interface pathGetter {
```

Comment:
Interfaces starting with a lower-case letter are [uncommon](https://github.com/basarat/typescript-book/blob/master/docs/styleguide/styleguide.md#interface)

Reply:
Maybe also move this interface next to the implementation of the PathGetter (= `ServerPath`) to be exported and reused from there? (It is duplicated at another place as well).
I personally would also be fine with having no interface and just using `ServerPath` as the type.

Reply:
Happy to rename it to start with upper-case 👍🏻 

For the interface - this is actually intentional and I don't see those as duplicates?

The `ServerPath` implementation implements _all_ of the methods/attributes and then the installer only requires subset of those, and `ClientHandler` requires different subset - hence they each have their own declaration of the interface depending on the needs.

Is this not how you'd use interfaces in TypeScript? It is a relatively common pattern in Go to reduce the exposure and limit responsibilities.

Reply:
That said its seems like a minor change with the size of the class anyway - and if I extract it into its own file then that addresses another concern I had - which is dependency cycles - so yeah - I can avoid interfaces altogether.

Reply:
Yeah, I suspected something like that. My personal style would've just been to keep it simple that's why I commented on that :) I did not notice that the subset differed – I just glanced over the naming and suspected them to be the same.
In the end I'm happy with either way 👍